test/utils/fixture.ts

```diff
@@ -10,20 +21,73 @@ export function setupStageWithFixture(
   stageName: string,
   fixtureName: string
 ): void {
-  const stagePath = path.join(rootDir, stageName);
-  shell.mkdir(stagePath);
+  const stagePath = getStagePath(stageName);
+  shell.mkdir('-p', stagePath);
   shell.exec(
     `cp -a ${rootDir}/test/${testDir}/fixtures/${fixtureName}/. ${stagePath}/`
   );
-  shell.ln(
-    '-s',
-    path.join(rootDir, 'node_modules'),
-    path.join(stagePath, 'node_modules')
-  );
+  if (getPackageManager() === 'npm') {
+    shell.ln(
+      '-s',
+      path.join(rootDir, 'node_modules'),
+      path.join(stagePath, 'node_modules')
+    );
+  }
   shell.cd(stagePath);
+  if (getPackageManager() === 'yarn2') {
+    // Setup stage directory as a project root for yarn2
+    fs.writeFileSync(
+      '.yarnrc.yml',
+      `yarnPath: ${path.join(
+        rootDir,
+        'test/yarn2/.yarn/releases/yarn-sources.cjs'
+      )}\ncacheFolder: ${rootDir}/test/yarn2/.yarn/cache`
+    );
+    // Copy seed lockfile with most resolutions already performed.  Makes tests faster.
+    shell.exec(`cp ${rootDir}/test/yarn2/yarn.lock ${stagePath}/`);
+  }
+  packageManagerInstall();
+}
+
+export function packageManagerInstall() {
+  if (getPackageManager() === 'yarn2') {
+    // Add tsdx dependency pointed to project directory
+    const pkg = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+    pkg.dependencies = pkg.dependencies ?? {};
+    pkg.dependencies['tsdx'] = `portal:${rootDir}`;
+    fs.writeFileSync('package.json', JSON.stringify(pkg, null, 2));
+    const { code } = shell.exec(`yarn`);
+    if (code !== 0) throw new Error('yarn install failed');
+  }
 }
 
 export function teardownStage(stageName: string): void {
   shell.cd(rootDir);
-  shell.rm('-rf', path.join(rootDir, stageName));
+  shell.rm('-rf', getStagePath(stageName));
+}
+
+export function getPackageManager() {
+  if (process.env.TSDX_TEST_PACKAGE_MANAGER === 'yarn2') return 'yarn2';
+  return 'npm';
+}
+
+export function getStagePath(stageName: string) {
+  return path.join(stageRootDir, stageName);
+}
+
+export function getLibTester(libPath: string) {
+  const absLibPath = path.resolve(libPath);
+  function evaluate(expression: string): any {
+    const isYarn2 = getPackageManager() === 'yarn2';
+    const libTesterPath = require.resolve('./lib-tester.js');
+    const args: libTester.JsonArgs = { libPath: absLibPath, expression };
+    const encodedValue = shell.exec(
+      `${isYarn2 ? 'yarn ' : ''}node ${libTesterPath} ${Buffer.from(
```

Comment:
IIRC I set it up so that jest is still running out of yarn 1 `node_modules` (less changes to the tests) and only `tsdx` and the build output of `tsdx` are executed with PnP.  `yarn node` is responsible for installing PnP into the node process by setting the right `NODE_OPTIONS` env var.

To support unit testing, we can switch to running everything under yarn 2.  I tried that earlier on in my experimenting, so I can switch back to it.